File: backend/src/main/java/at/ac/htl/features/shoppingcart/ShoppingCartDto.java
package at.ac.htl.features.shoppingcart;

import at.ac.htl.features.casing.Case;
import at.ac.htl.features.casing.CaseDto;
import at.ac.htl.features.cpu.CPUDto;
import at.ac.htl.features.cpuCooler.CpuCoolerDto;
import at.ac.htl.features.internalHarddrive.InternalHarddrive;
import at.ac.htl.features.internalHarddrive.InternalHarddriveDto;
import at.ac.htl.features.motherboard.MotherboardDto;
import at.ac.htl.features.gpu.GPUDto;
import at.ac.htl.features.powersupply.PowersupplyDto;
import at.ac.htl.features.ram.RAMDto;
import java.time.LocalDateTime;

public record ShoppingCartDto(
    Long id,
    CPUDto cpu,
    MotherboardDto motherboard,
    GPUDto gpu,
    RAMDto ram,
    CaseDto cases,
    PowersupplyDto powersupply,
    InternalHarddriveDto internalHarddrive,
    CpuCoolerDto cpuCooler,
    Double totalPrice,
    LocalDateTime createdAt,
    LocalDateTime updatedAt
) {
    
}

File: android_frontend/app/src/main/java/at/htl/leonding/android_frontend/data/model/ShoppingCartDto.kt
// FILE: data/model/ShoppingCartDto.kt
package at.htl.leonding.android_frontend.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

data class ShoppingCartDto(
    @Json(name = "id") val id: Long,
    @Json(name = "totalPrice") val totalPrice: Double? = 0.0,
    @Json(name = "cpu") val cpu: CpuDto?,
    @Json(name = "gpu") val gpu: GpuDto?,
    @Json(name = "ram") val ram: RamDto?,
    @Json(name = "motherboard") val motherboard: MotherboardDto?,
    @Json(name = "internalHarddrive") val storage: HarddriveDto?,
    @Json(name = "powersupply") val psu: PowerSupplyDto?,
    @Json(name = "cpuCooler") val cooler: CpuCoolerDto?
) {
    fun getItemCount(): Int {
        return listOf(cpu, gpu, ram, motherboard, storage, psu, cooler)
            .count { it != null }
    }
}
